src/clr-angular/utils/drag-and-drop/providers/drag-and-drop-event-bus.spec.ts

```
@@ -4,50 +4,60 @@
+        it("should broadcast from correct Observable on drag end event type", function() {
+            this.dndEventBus.dragEnded.subscribe((event: ClrDragEvent) => {
+                isEmitted = true;
+                expect(event.type).toBe(ClrDragEventType.DRAG_END);
             });
-            this.dndEventBus.notifyAllOnDragMove(testPayLoad);
+            this.dndEventBus.broadcast({type: ClrDragEventType.DRAG_END});
+            expect(isEmitted).toBeTruthy();
         });
 
-        it("notifyAllOnDragEnd() should cast to dragEnd subject", function() {
-            const testPayLoad: ClrDraggableEvent = {dragDataStore: "dragStartEnd"};
-            this.dndEventBus.dragEnded.subscribe((value: any) => {
-                expect(value).toBe(testPayLoad);
-                expect(value.dragDataStore).toBe("dragStartEnd");
+        it("should broadcast from correct Observable on drop event type", function() {
+            this.dndEventBus.dropped.subscribe((event: ClrDragEvent) => {
+                isEmitted = true;
+                expect(event.type).toBe(ClrDragEventType.DROP);
             });
-            this.dndEventBus.notifyAllOnDragEnd(testPayLoad);
+            this.dndEventBus.broadcast({type: ClrDragEventType.DROP});
+            expect(isEmitted).toBeTruthy();
         });
 
-        it("notifyAllOnDrop() should cast to drop subject", function() {
-            const testPayLoad: ClrDraggableEvent = {dragDataStore: "drop"};
-            this.dndEventBus.dropped.subscribe((value: any) => {
-                expect(value).toBe(testPayLoad);
-                expect(value.dragDataStore).toBe("drop");
+        it("should not broadcast anything on wrong event type", function() {
+            this.dndEventBus.dropped.subscribe((event: ClrDragEvent) => {
+                isEmitted = true;
             });
-            this.dndEventBus.notifyAllOnDrop(testPayLoad);
+            this.dndEventBus.broadcast({type: "if ever wrong type is passed"});
```

Wait, how does that even compile? Typescript should throw an error here. 😮

---

Yes 😂 that's why I was asking whether I should throw an exception or not. In the test files, it's compiling. But when editing other files in IDE, it notifies me by showing that squiggly error on the type.

---

doesnt this mean that something is wrong with our `tslint` config?

---

It's just that `this` isn't typed. But there isn't a real point to using the user context from Jasmine here, so we should just use simple variables, that way they'll be properly typed.